### PR TITLE
Add drawille

### PR DESCRIPTION
**Pull request by sshbio**
Emacs lisp version of [drawille](https://github.com/asciimoo/drawille) library, which permit to display graphics using braille characters.

My version can convert matrices or strings to drawille representations:
```
######                                              
#     # #####    ##   #    # # #      #      ###### 
#     # #    #  #  #  #    # # #      #      #      
#     # #    # #    # #    # # #      #      #####  
#     # #####  ###### # ## # # #      #      #      
#     # #   #  #    # ##  ## # #      #      #      
######  #    # #    # #    # # ###### ###### ###### 
```

`M-x drawille-buffer RET`

```
⡏⠉⠉⡆⡖⠒⢢⢀⠔⠢⡀⡆⠀⢰⢰⢰⠀⠀⠀⡆⠀⠀⢰⣒⣒⠂
⠧⠤⠤⠃⠏⠉⠣⠸⠉⠉⠇⠗⠉⠺⠸⠸⠤⠤⠄⠧⠤⠤⠸⠤⠤⠄
```

### CODE REVIEW

File: recipes/drawille.el [hunk 1]
```diff
@@ -0,0 +1,3 @@
+(drawille
+ :fetcher github
+ :repo "sshbio/elsip-drawille")
```
Comment (syohex) [inline recipes/drawille.el]: Typo `elsip-drawille` -> `sshbio/elisp-drawille`

**Comment by syohex** [comment]
You should fix following part. I cannot send PR because git treat your package as binary file. I suppose git misjudges by ascii art part.

## package header at first line

#### Original

https://github.com/sshbio/drawille/blob/master/drawille.el#L1

#### Fixed version

```lisp
;;; drawille.el --- Drawille implementation in elisp
```

## cl.el

Because your package uses cl-lib macros/functions instead of cl.el such as `cl-dotimes`, `cl-loop`.

#### Original

https://github.com/sshbio/drawille/blob/master/drawille.el#L35

#### Fixed version

```
(require 'cl-lib)
```

And add package dependency at header part.

```
;; Package-Requires: ((cl-lib "0.5"))
```

**Comment by sshbio** [comment]
Thank you much for these indications: This is my first package, and I did not already know how Melpa packages were built.

I did my best to make the package match these criteria.  Let see how the build occur...

About the repo name, any preference to name an elisp package (append ".el", "-el" prepend "emacs-", "lisp-"...)?

Thank you for your precious help!

**Comment by purcell** [comment]
Looks good to me. We don't have any preferences re. repo names - the current one is fine.

**Comment by sshbio** [comment]
Thank you @purcell!  Roger that.

File: recipes/drawille.el (hunk 1 above)
Comment (sshbio) [inline recipes/drawille.el]: I read the syntax guide, then I have no excuse! :stuck_out_tongue:

**Comment by wasamasa** [comment]
> Because your package uses cl-lib macros/functions instead of cl.el such as cl-dotimes, cl-loop.

This isn't quite correct.  `dotimes` is a built-in macro in `subr.el`.  `cl-dotimes` is a thing, but is only necessary if you need `cl-block` features, such as `cl-return` for breaking out of the loop.

**Comment by sshbio** [comment]
Thank you @wasamasa.

@syohex I will use cl-loop in a next version, so I will add this dependency (`cl` or `cl-lib`) all the same.

**Pull request merged**